src/EditorFeatures/CSharpTest/Diagnostics/CSharpCompilerDiagnosticAnalyzerProviderService.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Reflection;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Diagnostics.CSharp;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics
+{
+    [Export(typeof(IWorkspaceDiagnosticAnalyzerProviderService))]
+    internal class CSharpCompilerDiagnosticAnalyzerProviderService : IWorkspaceDiagnosticAnalyzerProviderService
```

Comment:
When testing, the `DiagnosticAnalyzerService` wasn't returning compiler diagnostics, which broke most of my tests. I insert this into the `ExportProvider` so that I get the diagnostics that the `AddImportFeatureService` expects.

https://github.com/dotnet/roslyn/blob/a4132792ff04f199d64a495b9fc2796cfbc7441b/src/EditorFeatures/CSharpTest/AddMissingImports/CSharpAddMissingImportsFeatureServiceTests.cs#L19-L27

Reply:
Why do all other codefix/refactoring tests work then?  This feels like we're needing to do extra work because we're not just usign the standard system we have for testing this stuff... :-/

Reply:
I have been pointed to a better way to gather the compiler diagnostics I am after. This will remove the need for this class and the ExportProvider manipulation.